### PR TITLE
Add v1.6 data types

### PR DESCRIPTION
**Pull request by proelke**

### CODE REVIEW

File: ocpp/v16/datatypes.py [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+
+@dataclass
+class AuthorizationData:
+    """
+    Elements that constitute an entry of a Local Authorization List update.
+    """
+
+    id_tag: IdToken
+    id_tag_info: IdTagInfo
```
Comment (OrangeTux) [inline ocpp/v16/datatypes.py:51]: This field is optional.

File: ocpp/v16/datatypes.py [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+@dataclass
+class IdTagInfo:
+    """
+    Contains status information about an identifier. It is returned in
+    Authorize, Start Transaction and Stop Transaction responses.
+
+    If expiryDate is not given, the status has no end date.
+    """
+
+    status: AvailabilityStatus
```
Comment (OrangeTux) [inline ocpp/v16/datatypes.py:39]: This type is AuthorizationStatus

**Review by OrangeTux — CHANGES_REQUESTED** [review]
I suggest to remove the `CiStringXXXType` and the `IdToken` All other classes model a type; a bag of attributes. The `CiStringXXXType` and `IdToken` are merely a string with some limits on their length. 

It also creates a non-friendly API. To read the value of the an `IdTagInfo`:

``` python
assert IdTagInfo.parent_id_tag.id_token.ci_string_20 == 'my-rfid'
```

Ideally, I want the API to be:

``` python
assert IdTagInfo.parent_id_tag == 'my-rfid'
```

I'm not sure yet, a nice way to impose this length on attributes of type `CiStringXXXType` and `IdToken`. So for now I suggest that the length is not validated and replace the type hints of attributes using those types with `str`. What do yo think?

**Comment by proelke** [comment]
I was debating adding them from the beginning. Then I debated adding some kind of custom validator, but I figured you might be against the complexity for such a small thing. To avoid a dependency I could use a __post_init__ to verify the length of the string in IdTagInfo and remove the CiStringXXXType. Or I can just drop it all together and leave it as is.

File: ocpp/v16/datatypes.py (hunk 1 above)
Comment (proelke) [reply ocpp/v16/datatypes.py]: Fixed.

File: ocpp/v16/datatypes.py (hunk 2 above)
Comment (proelke) [reply ocpp/v16/datatypes.py]: Fixed.

**Comment by OrangeTux** [comment]
> To avoid a dependency I could use a post_init to verify the length of the string in IdTagInfo and remove the CiStringXXXType

I like that solution

**Review by OrangeTux — CHANGES_REQUESTED** [review]
The code looks good. Make sure to fix the lint errors so the CI job passes.

**Comment by proelke** [comment]
@OrangeTux fixed up the linting issues

**Pull request merged**